apps/bfd-server/bfd-server-war/src/test/resources/openapi/headers.yaml

```diff
@@ -0,0 +1,162 @@
+components:
+  parameters:
+    BlueButton-OriginalQueryId-header:
+      in: header
+      name: BlueButton-OriginalQueryId
+      required: true
+      schema:
+        type: string
+        format: uuid
+      description: |
+        A unique ID (e.g. UUID) generated by the frontend system, per HTTP request to that frontend system. 
+        This ID SHALL be included in all log events for the frontend system. This allows requests to be 
+        traced across systems. For asynchronous jobs, users SHALL use the BlueButton-OriginalQueryId for 
+        the HTTP request that created/submitted/whatever the job.
+    BlueButton-OriginalQueryCounter-header:
+      in: header
+      name: BlueButton-OriginalQueryCounter
+      required: false
+      schema:
+        type: integer
+        minimum: 1
+      description: |
+        Start at 1 and increment for every request to BFD with the same BlueButton-OriginalQueryId value.
+    BlueButton-OriginalQueryTimestamp-header:
+      in: header
+      name: BlueButton-OriginalQueryTimestamp
+      required: false
+      schema:
+        type: string
+        format: date-time
+      description: | 
+        An ISO-8601 UTC timestamp representing (roughly) when the original request reached the frontend system.
```

Comment:
These were probably originally written in the context of BB2 frontend, but I'm not sure if every partner is going to be a frontend system; some are intermediaries to an actual user-facing frontend system. I'm thinking we actually care when the request made it to the requestor's system, be it a frontend or intermediary.

Might need to ask a partner how they are populating this today, if they are actually taking an upstream timestamp from the original frontend requestor or just filling this out when their system gets the request. If the latter, we might want to update the verbiage on the description in these headers.

Despite the number of words here this isn't a huge deal, just a clarity thing, and could be handled in the future.

Reply:
Most of the timestamps I saw in splunk are within a couple hundred millisecs from timestamp on the log entry so it appears more like the latter.